src/services/expense.service.js

```diff
@@ -0,0 +1,80 @@
+let expenses = [];
+
+const resetExpenses = () => {
+  expenses = [];
+};
+
+let nextId = 1;
+
+const getAllExp = (userId, categories, from, to) => {
+  let expList = expenses;
+
+  if (userId) {
+    expList = expList.filter((exp) => exp.userId === +userId);
+  }
+
+  if (categories) {
+    expList = expList.filter((exp) => categories.includes(exp.category));
+  }
+
+  if (from && to) {
+    expList = expList.filter((exp) => exp.spentAt >= from && exp.spentAt <= to);
+  }
+
+  if (to) {
+    expList = expList.filter((exp) => exp.spentAt <= to);
+  }
+
+  if (from) {
+    expList = expList.filter((exp) => exp.spentAt >= from);
```

Comment:
The filtering logic for 'from' and 'to' dates is redundant. The check for both 'from' and 'to' on lines 20-22 makes the separate checks on lines 24-30 unnecessary. Consider removing the separate checks if the combined check is sufficient.